frontend/src/app/matching/matching.component.ts

```diff
@@ -0,0 +1,58 @@
+import { Component } from '@angular/core';
+import { MatchingService } from '../_services/matching.service';
+import { NgForm } from '@angular/forms';
+import { StorageService } from '../_services/storage.service';
+import { timeoutWith, throwError } from 'rxjs';
+
+@Component({
+  selector: 'app-matching',
+  templateUrl: './matching.component.html',
+  styleUrls: ['./matching.component.css']
+})
+export class MatchingComponent {
+
+  match: any = undefined
+
+  requested: boolean = false
+
+  queueLength: number = -1
+
+  constructor(private matchingService: MatchingService, private storageService: StorageService){}
+
+  ngOnInit() {
+    this.getQueueLength();
+  }
+
+  getMatch(userDetails: any) {
+    this.matchingService.enqueue(userDetails).pipe(timeoutWith(2000, throwError(() => {
+      this.matchingService.dequeue(userDetails["userid"]).subscribe((res) => {
```

Comment:
So is this subscribe stuff the rxjs?

Reply:
So basically it's a socket I think that waits for an HTTP response from the URL. It does not timeout unless you specify it like I did here.